command.go

```diff
@@ -1666,58 +1663,57 @@ func (c *Command) LocalNonPersistentFlags() *flag.FlagSet {
 }
 
 // LocalFlags returns the local FlagSet specifically set in the current command.
+// This function does not modify the flags of the current command, it's purpose is to return the current state.
 func (c *Command) LocalFlags() *flag.FlagSet {
 	c.mergePersistentFlags()
 
-	if c.lflags == nil {
```

Comment:
The use of this member variable is for efficiency. We probably don’t want to remove it

Reply:
Hi Marc, thank you for taking a look!
Does this optimization really matter? I can't imagine any code calling `LocalFlags` in a loop or something like that.
It seems to me that these fields are confusing as they appear to represent command state rather than to cache internal state of some functions.
If you prefer to keep this optimisation, do you agree to rename it `lflags` to something like `lflagsTmpState`?

Reply:
The `lflags` field was introduced https://github.com/spf13/cobra/commit/5c9146990baf14f44c4c470befa56c7ee5b62130 and its initial purpose is not clear to me.

Reply:
Hi @marckhouzam do you really want me to keep these fields? If so are you OK with renaming them as I suggested before? Thanks!

Reply:
Hi @marckhouzam, will you have time to look at this PR? What would be the best way to move forward? Thanks!

Reply:
Hello @marckhouzam is there any chance you can find some time to review this PR (clarify the documentation of the `LocalFlags` function? Thanks!

Reply:
@niamster I found this commit which introduced `iflags` and it seems to indicate there is an improvement in performance: https://github.com/spf13/cobra/commit/6202b5942b8f31501d14d8cb192790d8a10302f4

Although your code changes look safe, I prefer not to change things without a good reason;  cobra is being used by so many programs that we need to prioritize stability.

For example, what if a program stored the pointer returned by `InheritedFlags()` for some reason.  With this PR a second call to `InheritedFlags()` would return a different pointer.  Would it matter?  Probably not, but I cannot be sure.  So, let's not invite some unexpected problem 😄 

If you want to limit the PR to the comment changes, that would be fine.

Reply:
Hi Marc, thank you for finding some time to look into this issue. OK, yes I understand your concern. I'll try to find some time next week to limit this PR to comment changes only 😄

Reply:
Hi @marckhouzam I'm sorry for the delay. I've updated this PR to address the documentation only, can you please take a look? Thanks!